Mathlib/CategoryTheory/Sites/Coherent/SheafComparison.lean

```diff
@@ -0,0 +1,220 @@
+    refine ⟨fun ⟨⟩ ↦ ⟨()⟩, ?_⟩
+    rintro ⟨⟩
+    simp
+  · rw [← effectiveEpi_iff_effectiveEpiFamily]
+    infer_instance
+
+theorem exists_effectiveEpiFamily_iff_mem_induced (X : C) (S : Sieve X) :
+    (∃ (α : Type) (_ : Finite α) (Y : α → C) (π : (a : α) → (Y a ⟶ X)),
+      EffectiveEpiFamily Y π ∧ (∀ a : α, (S.arrows) (π a)) ) ↔
+    (S ∈ F.inducedTopologyOfIsCoverDense (coherentTopology _) X) := by
+  refine ⟨fun ⟨α, _, Y, π, ⟨H₁, H₂⟩⟩ ↦ ?_, fun hS ↦ ?_⟩
+  · apply (mem_sieves_iff_hasEffectiveEpiFamily (Sieve.functorPushforward _ S)).mpr
+    refine ⟨α, inferInstance, fun i => F.obj (Y i),
+      fun i => F.map (π i), ⟨?_,
+      fun a => Sieve.image_mem_functorPushforward F S (H₂ a)⟩⟩
+    exact F.map_finite_effectiveEpiFamily _ _
+  · obtain ⟨α, _, Y, π, ⟨H₁, H₂⟩⟩ := (mem_sieves_iff_hasEffectiveEpiFamily _).mp hS
+    refine ⟨α, inferInstance, ?_⟩
+    let Z : α → C := fun a ↦ (Functor.EffectivelyEnough.presentation (F := F) (Y a)).some.p
+    let g₀ : (a : α) → F.obj (Z a) ⟶ Y a := fun a ↦ F.effectiveEpiOver (Y a)
+    have : EffectiveEpiFamily _ (fun a ↦ g₀ a ≫ π a) := inferInstance
+    refine ⟨Z , fun a ↦ Full.preimage (g₀ a ≫ π a), ?_, fun a ↦ (?_ : S.arrows (Full.preimage _))⟩
+    · refine F.finite_effectiveEpiFamily_of_map _ _ ?_
+      simpa using this
+    · obtain ⟨W, g₁, g₂, h₁, h₂⟩ := H₂ a
+      rw [h₂]
+      have : Full.preimage (g₀ a ≫ g₂ ≫ F.map g₁) = (Full.preimage (g₀ a ≫ g₂)) ≫ g₁ := by
+        apply Faithful.map_injective (F := F)
+        simp
+      rw [this]
+      exact S.downward_closed h₁ _
+
+lemma eq_induced : haveI := F.reflects_precoherent
+    coherentTopology C =
+      F.inducedTopologyOfIsCoverDense (coherentTopology _) := by
+  ext X S
+  have := F.reflects_precoherent
+  rw [← exists_effectiveEpiFamily_iff_mem_induced F X]
+  rw [← coherentTopology.mem_sieves_iff_hasEffectiveEpiFamily S]
+
+lemma coverPreserving : haveI := F.reflects_precoherent
+    CoverPreserving (coherentTopology _) (coherentTopology _) F := by
+  rw [eq_induced F]
+  apply LocallyCoverDense.inducedTopology_coverPreserving
+
+instance coverLifting : haveI := F.reflects_precoherent
+    F.IsCocontinuous (coherentTopology _) (coherentTopology _) := by
+  rw [eq_induced F]
+  apply LocallyCoverDense.inducedTopology_isCocontinuous
+
+instance isContinuous : haveI := F.reflects_precoherent
+    F.IsContinuous (coherentTopology _) (coherentTopology _) :=
+  Functor.IsCoverDense.isContinuous _ _ _ (coverPreserving F)
+
+section SheafEquiv
+
+universe u
+
+variable {C D : Type (u+1)} [LargeCategory C] [LargeCategory D] (F : C ⥤ D)
+  [F.PreservesFiniteEffectiveEpiFamilies] [F.ReflectsFiniteEffectiveEpiFamilies]
+  [F.Full] [F.Faithful]
+  [Precoherent D]
+  [F.EffectivelyEnough]
+
+/--
+The equivalence from coherent sheaves on `Stonean` to coherent sheaves on `CompHaus`
+(i.e. condensed sets).
+-/
+noncomputable
+def equivalence (A : Type _) [Category.{u+1} A] [HasLimits A] : haveI := F.reflects_precoherent
```

Comment:
Can you mention the three equivalence in the doc at beginning, as main applications?